internal/worker/cache.go

```diff
@@ -273,6 +277,25 @@ func (c *cache) handleContractRenew(event api.EventContractRenew) {
 	c.cache.Set(cacheKeyDownloadContracts, contracts)
 }
 
+func (c *cache) handleHostUpdate(e api.EventHostUpdate) {
+	// return early if the cache doesn't have contracts
+	value, found, _ := c.cache.Get(cacheKeyDownloadContracts)
+	if !found {
+		return
+	}
+	contracts := value.([]api.ContractMetadata)
+
+	// update the host's IP in the cache
+	for i, contract := range contracts {
+		if contract.HostKey == e.HostKey {
+			contracts[i].HostIP = e.NetAddr
+			break
```

Comment:
If we have multiple contract sets it might be possible that we have more than 1 contract with a host, right? Might be safer to not break here.